src/vsr/journal.zig

```diff
@@ -1724,6 +1730,7 @@ pub fn JournalType(comptime Replica: type, comptime Storage: type) type {
         }
 
         pub fn set_header_as_dirty(journal: *Journal, header: *const Header.Prepare) void {
+            const replica: *Replica = @alignCast(@fieldParentPtr("journal", journal));
```

Comment:
Perhaps it would be better to pass `cluster` as a parameter to `set_header_as_dirty`, or even to journal's init? Reaching out to replica like this feels inelegant.

Reply:
:bulb:  Aha, we can just use `header.cluster`!